File: pdftract-dotnet/src/Pdftract/Models/Metadata.cs
using MessagePack;
using System.Text.Json.Serialization;

namespace Pdftract.Models;

/// <summary>
/// Represents PDF document metadata (title, author, dates, etc.).
/// </summary>
[GenerateSerializer]
public record Metadata
{
    [JsonPropertyName("title")]
    public string? Title { get; init; }

    [JsonPropertyName("author")]
    public string? Author { get; init; }

    [JsonPropertyName("subject")]
    public string? Subject { get; init; }

    [JsonPropertyName("keywords")]
    public string? Keywords { get; init; }

    [JsonPropertyName("creator")]
    public string? Creator { get; init; }

    [JsonPropertyName("producer")]
    public string? Producer { get; init; }

    [JsonPropertyName("created_date")]
    public DateTime? CreatedDate { get; init; }

    [JsonPropertyName("modified_date")]
    public DateTime? ModifiedDate { get; init; }
}

File: pdftract-java/src/main/java/com/jedarden/pdftract/Metadata.java
package com.jedarden.pdftract;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Document metadata.
 */
public record Metadata(
    @JsonProperty("page_count") int pageCount,
    @JsonProperty("title") String title,
    @JsonProperty("author") String author,
    @JsonProperty("creator") String creator,
    @JsonProperty("has_xmp") Boolean hasXmp
) {}
